state-chain/runtime/src/migrations/solana_remove_unused_channels_state.rs

```diff
@@ -0,0 +1,78 @@
+use crate::*;
+use frame_support::{pallet_prelude::Weight, traits::OnRuntimeUpgrade};
+
+use codec::{Decode, Encode};
+use pallet_cf_elections::{
+	electoral_systems::composite::tuple_7_impls::CompositeElectoralUnsynchronisedStateMapKey,
+	ElectoralUnsynchronisedStateMap,
+};
+use sp_core::bounded::alloc::collections::BTreeSet;
+#[cfg(feature = "try-runtime")]
+use sp_runtime::DispatchError;
+
+pub struct SolanaRemoveUnusedChannelsState;
+
+impl OnRuntimeUpgrade for SolanaRemoveUnusedChannelsState {
+	fn on_runtime_upgrade() -> Weight {
+		let addresses_in_use: BTreeSet<_> = pallet_cf_ingress_egress::DepositChannelLookup::<
+			Runtime,
+			SolanaInstance,
+		>::iter_keys()
+		.chain(
+			pallet_cf_ingress_egress::DepositChannelPool::<Runtime, SolanaInstance>::iter_values()
+				.map(|deposit_channel| deposit_channel.address),
+		)
+		.collect();
+
+		let keys = ElectoralUnsynchronisedStateMap::<Runtime, SolanaInstance>::iter_keys()
+			.collect::<Vec<_>>();
+
+		log::info!("🧮 Number of Solana Addresses stored before migration: {:?}", keys.len());
+		let mut removed = 0;
+		for key in keys.into_iter() {
+			if let CompositeElectoralUnsynchronisedStateMapKey::C((address, _asset)) = key {
+				if !addresses_in_use.contains(&address) {
+					ElectoralUnsynchronisedStateMap::<Runtime, SolanaInstance>::remove(key);
+					removed += 1;
+				}
+			}
+		}
+		log::info!("🚮 Number of addresses deleted: {:?}", removed);
+		if removed == 0 {
+			log::warn!("❗️ No Solana Addresses deleted - consider removing this migration.");
+		}
+
+		Weight::zero()
+	}
+
+	#[cfg(feature = "try-runtime")]
+	fn pre_upgrade() -> Result<Vec<u8>, sp_runtime::TryRuntimeError> {
```

Comment:
nit, putting pre_upgrade before the `on_runtime_upgrade` makes it read sequentially